cmake/macros/macro_deal_ii_add_definitions.cmake

```diff
@@ -26,6 +27,14 @@ macro(deal_ii_add_definitions _name)
     set_property(TARGET ${_name}_${_build_lowercase}
       APPEND PROPERTY COMPILE_DEFINITIONS "${ARGN}"
       )
+
+   # For release builds (and their corresponding object files),
+   # use interprocedural optimizations if possible
+   if (("${_build}" STREQUAL "RELEASE") AND ("${DEAL_II_COMPILER_SUPPORTS_IPO}" STREQUAL "YES"))
+     set_property(TARGET ${_name}_${_build_lowercase}
+                  PROPERTY INTERPROCEDURAL_OPTIMIZATION TRUE)
+     message(STATUS "Setting LTO properties on ${_name}_${_build_lowercase} for build type ${_build}")
+   endif()
```

Comment:
This needs to go into `macro_populate_target_properties.cmake` (which defines the `populate_target_properties` macro that is called for every object and library target we define).